### ISSUE TITLE
Running multiple concurrent exports on one index

### ISSUE THREAD

**Issue by shardnit** [jprante/elasticsearch-knapsack#69]
I am working on a service that collects all the results of user-supplied query. I think knapsack is a much better candidate for this kind of thing. Instead of writing my own service I could use knapsack with different path names (for each user). Anything I should keep in mind to keep this functionality fairly scalable (using knapsack)?
Also, another pain point is that every document becomes a separate file in the export (uncompressed). Any recommended approach to combine all the results in a single (export) file efficiently (since there could be millions of result documents)?

**Comment by jprante** [comment]
I do not think that knapsack is a good user-oriented service. The archive formats are not for reuse in contexts other than knapsack because there are conventions that are quite specific. Knapsack is designed for admin export/import only and exporting should not be run when indexing is active for example. You can run as many exports as you want in parallel to multiple archives but the result can be trusted only if the indices which are exported are "quiet".

If you use bulk format, all documents will be combined into a single file (json lines style).

**Comment by shardnit** [comment]
Thanks for your feedback Jorg. Pressing "close" button since this is not really an issue :). 
I think in my case its more or less guaranteed that while an export is going on there wont be any new documents imported to that index. I think I am gonna give knapsack a go for this service. Atleast with this I wouldn't have to transfer the data over HTTP wire.
Another question - Is it possible to collect some basic stats about an export like how much time it took etc.? Also, I see a mention of S3 integration in the readme but couldn't find anything on its usage.

**Comment by shardnit** [comment]
Another issue @jprante. The `_export/state` endpoint works reliably on my localhost ES instance, but always returns count 0 on production (even when there are ongoing exports). I have installed this plugin only on (one) query node (and not on the data nodes). Could this be causing problems?

**Comment by jprante** [comment]
`_export/state` may only work on a data node.  Haven't tested on data-less nodes, so consider this a bug.

The debug log level might give more insight information about the knapsack progress. There is no perfect monitoring solution yet.

S3 was a community contribution, I removed it for now because I do not use S3 and have no time to test S3 API. All in all it was just a tiny piece of code to execute an S3 upload command on the export file and delete it afterwards, while S3 import was an S3 download followed by an import. The S3 plugin zip was extended by S3 API SDK.
